providers.py

```diff
@@ -177,7 +177,7 @@ def msg_push(self, body, repo):
             last_commit_url = body['commits'][-1]['url']
             commit_messages = [
                 dict(msg=c['message'][:80].split('\n')[0], hash=c['id'][:8],
-                     url=c['url']) for c in body['commits']
+                     url=c['url'][:-28]) for c in body['commits']
```

Comment:
This URL is always of the form

https://host/user/project/commit/50d7590db426ce821148af769ea18d556e97f393

The hash is very long (sha1 hash, 40 chars) and uselessly clutters the message.

The URL works with smaller hashes, as long as it’s unique (in the same way that git itself handles it).

1. It is supposed to shorten the hash.
2. This works on any gitlab instance, as long as git uses sha1 (so, forever?)
3. Nope, only the end of the sha1.

For example see these two links:
https://lab.louiz.org/louiz/biboumi/commit/50d7590db426ce821148af769ea18d556e97f393

https://lab.louiz.org/louiz/biboumi/commit/50d7590db426

I like the second one best.